migrations/2024-04-17-135931_default-versions-table/up.sql

```diff
@@ -0,0 +1,24 @@
+create table default_versions
+(
+    crate_id   integer not null
+        constraint default_versions_pk
+            primary key
+        constraint default_versions_crates_id_fk
+            references crates
+            on delete cascade,
+    version_id integer not null
+        constraint default_versions_versions_id_fk
+            references versions
+            on delete no action deferrable initially deferred
```

Comment:
I'm thinking about the edge case of deleting a version where there's only a single version in a crate. Would the crate also be deleted?
If it's not, since the `version_id` column is not null in this PR, what will happen in this case?

Reply:
it would fail to delete the version. and I guess that's a good thing because then the admin knows that they should've deleted the whole crate instead.